pkg/jobs/utils/helpers.go

```diff
@@ -465,3 +496,40 @@ func parseVersionInfo(msg string) (channel, upstream string, semversion semver.V
 
 	return channel, upstream, semversion, nil
 }
+
+func GetClusterType(
+	hiveset hiveclient.Interface,
+	dc dynamic.Interface,
+	clusterName string,
+	clusterNamespace string) (string, error) {
+	// shortcut for Hypershift cluster
+	if clusterName != clusterNamespace {
+		return HypershiftClusterType, nil
+	}
+
+	// if clusterName and clusterNamespace are equal we need more info
+	cluster, err := hiveset.HiveV1().ClusterDeployments(clusterName).Get(context.TODO(), clusterName, v1.GetOptions{})
+	if err == nil && cluster != nil {
+		return StandaloneClusterType, nil
+	}
+
+	hostedCluster, hcErr := dc.Resource(HCGVR).Namespace(clusterNamespace).Get(
+		context.TODO(), clusterName, v1.GetOptions{})
+	if hcErr == nil && hostedCluster != nil {
+		return HypershiftClusterType, nil
+	}
+
+	return StandaloneClusterType, hcErr
```

Comment:
If you do the above suggestions then in some cases this might return a hostedcluster not found error but maybe we should just return a new error that clearly state something like: failed to determine the cluster type because hive cp not found and hostedcluster not found